src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] innerArray = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size == innerArray.length) {
+            grow();
+        }
+        innerArray[size++] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index >= 0 && index <= size) {
+            T[] arr = (T[]) new Object[size - index];
+            if (size > index) {
+                System.arraycopy(innerArray, index, arr, 0, size - index);
+            }
+            innerArray[index] = value;
+            if (size + 1 > innerArray.length) {
+                grow();
+            }
```

Comment:
The capacity check and potential call to `grow()` should occur before inserting the new element at the specified index. This ensures that there is enough space in the array before shifting elements.